editoast/src/models/paced_train.rs

```diff
@@ -181,23 +181,26 @@ impl PacedTrain {
     pub fn iter_occurrences(&self) -> impl Iterator<Item = (TrainId, TrainSchedule)> {
         let mut base_occurrences = self.get_base_occurrences();
 
-        let enabled_modified_exceptions = self
+        let modified_exceptions = self
             .exceptions
             .iter()
-            .filter(|ex| !ex.disabled)
             .filter_map(|e| match e.exception_type {
                 ExceptionType::Modified { occurrence_index } => Some((occurrence_index, e)),
                 _ => None,
             });
 
-        for (occurrence_index, exception) in enabled_modified_exceptions {
+        for (occurrence_index, exception) in modified_exceptions {
             if let Some(occurrence) = base_occurrences.get_mut(occurrence_index as usize) {
-                let train_id = TrainId::PacedTrainModifiedException {
-                    paced_train_id: self.id,
-                    index: occurrence_index as u64,
-                    exception_key: exception.key.clone(),
-                };
-                *occurrence = (train_id, self.apply_exception(exception));
+                if exception.disabled {
+                    base_occurrences.remove(occurrence_index as usize);
```

Comment:
It can be very inefficient. The worst case is when all trains are disabled. `O(n²)`.
To improve this we can define a variable like: `let mut to_remove: vec![false, base_occurrences.len()];`.
Then, for each occurrence to remove, we set the corresponding value to `true`.
At the end of the loop we can do something like:

```rust
let occurences = base_occurrences.into_iter().zip(to_remove).filter_map(|(occ, to_remove)|
  if to_remove { None }
  else { Some(occ) }
).collect_vec();
```